### ISSUE TITLE
D:\KPWorkSpace\AllFireWall\SetupFireWall\Product.wxs(63,0): error LGHT0103: The system cannot find the file 'C:\WINDOWS\System32\vcruntime140_1.dll'.

### ISSUE THREAD

**Issue by Cldndan** [wixtoolset/issues#7516]
## Triage

D:\KPWorkSpace\AllFireWall\SetupFireWall\Product.wxs(63,0): error LGHT0103: The system cannot find the file 'C:\WINDOWS\System32\vcruntime140_1.dll'.

## Discussions and support

If you think that you are a developer approaching Wix for the first time, you will understand the query method below...
Accordingly, we ask for your understanding in advance.

Also, unfortunately, I'm not fluent in English, so I'd like to inform you that it can be a bit unnatural because of Google Translate.

1. Below is my packing list.
2. C:\WINDOWS\System32\vcruntime140_1.dll
the above file
"vcruntime140.dll" is on the same path,
Check result with property window
As a result of checking the list of accessible users of the file and the contents of the permission settings for each individual user, they are the same.

Nevertheless, I'm getting the same error as the title.

I'd like to know what the reason is. Actually I don't understand.

**Comment by barnson** [comment]
You're probably running from a 32-bit process.

**Comment by Cldndan** [comment]
First of all thank you for your reply.

1. I didn't compile for 32 bit.
It was compiled in 64-bit.

2. The C:\WINDOWS\System32\vcruntime140_1.dll file is, as far as I know, an extended version of C:\WINDOWS\System32\vcruntime140.dll technical functions, etc.
And it has already been confirmed that the same path and the same permissions as vcruntime140.dll are set.
- I also have the latest version of the redistributable package installed, version 14.36.32532.
- C:\Windows\SysWOW64\vcruntime140(d).dll and vcruntime140_clr0400.dll are found.
However, vcruntime140_1.dll was not found.
- This is confirmed after installing the latest version of the redistribution package, version 14.36.32532.
- Accordingly, C:\Windows\SysWOW64\vcruntime140_1.dll Not Found Error is an understandable error message, and if the above error is a 32-bit process error, the path display is incorrect. "C:\Windows\ SysWOW64\vcruntime140_1.dll", so it must be a bug.

3. If the Wix builder is 32-bit, I may have installed it incorrectly.
- However, since the current system is 64-bit Windows, the corresponding path is still displayed incorrectly.
- The route display must be displayed according to the current system.

- Downloaded from your company's official download site.
However, I was not sure if it was a 64-bit process download file, and if the file can be distributed free of charge, can you tell me the URL?
If it is a free distribution, please let me know.
4. Please tell me how to check if the Wix builder is 32/64 bit.

**Comment by Cldndan** [comment]
I query the following to be a bit more clear.

1. Is the wix builder of the Wix Toolset that you are currently distributing building and distributing in 32-bit?
- We know that it is compatible with both 32/64 bit OS (Windows).
- However, in "barnson"'s answer above, it is not clear whether the subject of "You're probably running from a 32-bit process." refers to the product I developed or the Wix builder, so I ask again.
2. If the above answer refers to the Wix builder, that is, because it is a 32-bit processor, then the dll file I specified and the dll file I specified on the path (the paths of all the dlls I specified are the same) should have the same error.
- If you have already detected this because 64-bit dlls cannot be loaded in my development product developed in 32-bit, it is because all dlls on the path I specified are 64-bit.
(As far as I know, my OS is 64 non-Wondows 11, of course, the same error should occur, but only one "vcruntime140_1.dll" file causes that error.)
3. If the Wix builder is 32-bit, it is difficult to understand that only certain files are causing the error.
-However, if you separately manage the supported DLL list of MS or Visual Studio 2022 supported on the current Wix Ver 3.1.2 (latest distribution) version, the file will not be supported. A more sensible warning message would be to output it as a warning message encouraging you to deploy it as a redistributable package.

- Accordingly, the current error message (file not found) is somewhat difficult to recognize as the correct message when looking at various issues and answers.
- This is considered a minor but serious bug in how to communicate.
What do you think?

**Comment by barnson** [comment]
Windows does 32-bit/64-bit file-system redirection behind the scenes.

**Comment by Cldndan** [comment]
First of all, you should read my query considering that it is an English version translated by Google Translator.

Because it may be delivered differently from my intention,
 It seems that you (barnson) did not give me an accurate answer.
  
I put a lot of effort into completing the sentence, but since my communication wasn't done well, I'm asking again.

Seriously, I'm asking again.

Regarding the error message below, I will inform you of my current system status. "D:\KPWorkSpace\AllFireWall\SetupFireWall\Product.wxs(63,0): error LGHT0103: The system cannot find the file 'C:\WINDOWS\System32\vcruntime140_1.dll'."

1. The current OS is Windows 11 64-bit OS.

2. There are several dll file names that start with "vcruntime140" under the C:\WINDOWS\System32 directory. Among them,
It was confirmed that vcruntime140_1.dll exists.
3. The files were installed in the process of installing the Visual Studio 2022 community version, as well as the 64-bit redistributable package (VC_redist.x64.exe) and the 32-bit redistributable package (VC_redist.x86.exe).
 4. As far as I know,
The files that are distributed and installed through the process of 3 above... that is, 64-bit Windows 11 OS
 
All dll files (including all dll files starting with vcruntime140) installed in the sub-path of "C:\WINDOWS\System32" are considered as 64-bit dll files, and 32-bit dll files are installed in the sub-path of "C:\Windows\SysWOW64". (including all dll files starting with vcruntime140) I'm judging by the files.

5. I know that the Wix builder is compatible on windows 11 64 bit even though it is distributed as 32 bit.
- That's why I don't understand why the corresponding error message is output.
- If the corresponding vcruntime140_1.dll file is not supported by the current Wix builder executable, it is asserting that it would be correct to indicate that it is not supported.
- However, since the file exists and the permission settings are set the same as other dll files, it is argued that it should be output as a warning message rather than an error message.
(The vcruntime140_1.dll file is currently not supported by Wix Builder.
Accordingly, it cannot be loaded or mounted to the MSI. It is argued that it should be changed to this expression.)
- And it is argued that it can be distributed using a redistribution package, and a warning message with a neutral expression that there will be no problem in using the dll file and the functions included in the dll is sufficient.
- I am querying because I do not understand the contents of the message that the file does not exist even though the file exists, and it is not judged to be a correct error message.
- I dared to judge that it is correct to register it as a Wix bug and issue, so I ask for an answer.

**Comment by barnson** [comment]
If file system redirection is happening, the file would be reported as not found, so the WiX error message would be correct. You might check using a tool like Process Monitor to determine the underlying API failure.

**Comment by Cldndan** [comment]
Yes, I understand.
In the end, the Wix builder was released as built for 32-bit,
C:/Windows/System32, that is, when the dll designated as the system path is loaded into the MSI, the Wix builder is recognized as 32-bit by the Windows File System, and redirection has occurred.

So, since it was redirected to C:/Windows/SysWow64, this problem occurs because the vcruntime140_1.dll file does not exist in the redirected path.

If so, the software I developed was built as 64-bit. So I need 64 bit.
So, 64-bit dlls must be distributed.
So it makes sense to choose one of the following two methods.

1. Copy the required 64-bit dll to a path different from the Windows System (C:/Windows/System32) path (for example, the same path as the executable file of the software I developed), and specify the copied path.

Then, because it is not a redirect target path by Windows File System,
The 64-bit dll file will be loaded into the MSI file.


2. How to clone a Wix project (including source) from Git and rebuild it with Visual Studio

The above two methods would be reasonable.

Thank you for your help in getting us to a sensible result.
